src/device-state/current-state.ts

```diff
@@ -130,100 +140,119 @@ const getStateDiff = (): DeviceStatus => {
-							storage_block_device: null,
-							cpu_temp: null,
-							cpu_id: null,
-					  }),
-				...(await sysInfo.getSystemChecks()),
-			};
-
-			stateForReport.local = {
-				...stateForReport.local,
-				...currentDeviceState.local,
-				...info,
-			};
-			stateForReport.dependent = {
-				...stateForReport.dependent,
-				...currentDeviceState.dependent,
-			};
-
-			// Report current state
-			await report();
-			// Finishing pending report
-			reportPending = false;
-		} catch (e) {
-			eventTracker.track('Device state report failure', { error: e });
-			// We use the poll interval as the upper limit of
-			// the exponential backoff
-			const maxDelay = await config.get('appUpdatePollInterval');
-			const delay = Math.min(
-				2 ** stateReportErrors * MINIMUM_BACKOFF_DELAY,
-				maxDelay,
-			);
-
-			++stateReportErrors;
-			await Bluebird.delay(delay);
-			reportCurrentState();
-		}
+		const currentDeviceState = await deviceState.getStatus();
+		// If hardwareMetrics is false, send null patch for system metrics to cloud API
+		const info = {
+			...(hardwareMetrics
+				? await sysInfo.getSystemMetrics()
+				: {
+						cpu_usage: null,
+						memory_usage: null,
+						memory_total: null,
+						storage_usage: null,
+						storage_total: null,
+						storage_block_device: null,
+						cpu_temp: null,
+						cpu_id: null,
+				  }),
+			...(await sysInfo.getSystemChecks()),
+		};
+
+		stateForReport.local = {
+			...stateForReport.local,
+			...currentDeviceState.local,
+			...info,
+		};
+		stateForReport.dependent = {
+			...stateForReport.dependent,
+			...currentDeviceState.dependent,
+		};
+
+		const stateDiff = getStateDiff();
+
+		// report state diff
+		throttledReport(
+			{ stateDiff, conf },
+			() => {
+				// If the patch succeeds update lastReport and reset the error counter
+				_.assign(lastReportedState.local, stateDiff.local);
+				_.assign(lastReportedState.dependent, stateDiff.dependent);
+				stateReportErrors = 0;
+
+				reportPending = false;
+			},
+			(e) => {
+				if (e instanceof StatusError) {
+					// We don't want these errors to be classed as a report error, as this will cause
+					// the watchdog to kill the supervisor - and killing the supervisor will
+					// not help in this situation
+					log.error(
+						`Non-200 response from the API! Status code: ${e.statusCode} - message:`,
+						e.message,
+					);
+
+					// We want to backoff on all errors, but without having the healthchecks
+					// get triggered.
+					// This will use retryAfter as maxDelay if the header is present in the
+					// response by the API
+					backoff(
+						reportCurrentState,
+						// Do not do exponential backoff if the API reported a retryAfter
+						e.retryAfter ? 0 : attempts,
+						maxDelay,
+						// Start the polling at the value given by the API if any
+						e.retryAfter ?? MINIMUM_BACKOFF_DELAY,
+					);
+				} else {
+					eventTracker.track('Device state report failure', { error: e });
```

Comment:
if we change the following line we can turn

```
[event]   Event: Device state report failure {"error":{"message":""}}
```

into

```
[event]   Event: Device state report failure {"error":"getaddrinfo ENOTFOUND api.xgho.st"}
```

```suggestion
					eventTracker.track('Device state report failure', { error: e.message });
```